src/ophyd_async/epics/pmac/_pmacTrajectory.py

```diff
@@ -0,0 +1,154 @@
+import time
+
+from bluesky.protocols import Flyable, Preparable
+from scanspec.specs import Frames
+
+from ophyd_async.core.async_status import AsyncStatus, WatchableAsyncStatus
+from ophyd_async.core.signal import observe_value
+from ophyd_async.core.utils import WatcherUpdate
+from ophyd_async.epics.motion import Motor
+from ophyd_async.epics.pmac import Pmac
+
+TICK_S = 0.000001
+
+
+class PmacTrajectory(Pmac, Flyable, Preparable):
+    """Device that moves a PMAC Motor record"""
+
+    def __init__(self, prefix: str, cs: int, name="") -> None:
+        # Make a dict of which motors are for which cs axis
+        self._fly_start: float
+        self.cs = cs
+        super().__init__(prefix, cs, name=name)
+
+    async def _ramp_up_velocity_pos(
+        self, velocity: float, motor: Motor, end_velocity: float
+    ):
+        # Assuming ramping to or from 0
+        max_velocity_acceleration_time = await motor.acceleration_time.get_value()
+        max_velocity = await motor.max_velocity.get_value()
+        delta_v = abs(end_velocity - velocity)
+        accl_time = max_velocity_acceleration_time * delta_v / max_velocity
+        disp = 0.5 * (velocity + end_velocity) * accl_time
+        return [disp, accl_time]
+
+    @AsyncStatus.wrap
+    async def prepare(self, stack: list[Frames[Motor]]):
+        # Which Axes are in use?
+
+        scanSize = len(stack[0])
```

Comment:
We should create a `path = Path(stack)` and call `chunk = path.consume(n)` to consume N points from the scan to process. In this case `chunk = path.consume() # consume the whole scan` and `scan_size = len(chunk)`